sys/net/application_layer/nanocoap/sock.c

```diff
@@ -77,57 +86,102 @@ static int _send_ack(nanocoap_sock_t *sock, coap_pkt_t *pkt)
     return sock_udp_send(sock, &ack, sizeof(ack), NULL);
 }
 
+static bool _id_or_token_missmatch(const coap_pkt_t *pkt, unsigned id,
+                                   const void *token, size_t token_len)
+{
+    switch (coap_get_type(pkt)) {
+    case COAP_TYPE_RST:
+    case COAP_TYPE_ACK:
+        return coap_get_id(pkt) != id;
+    default:
+        if (pkt->token == NULL) {
+            return token != NULL;
+        }
+        return memcmp(pkt->token, token, token_len);
+    }
+}
+
+static uint32_t _deadline_from_interval(uint32_t interval)
+{
+    return ztimer_now(ZTIMER_USEC) + interval;
+}
+
+static uint32_t _deadline_left_us(uint32_t deadline)
+{
+    uint32_t now = ztimer_now(ZTIMER_USEC);
+    if (now > deadline) {
+        return 0;
+    }
+
+    return deadline - now;
+}
```

Comment:
How about using `ms` everywhere?

Reply:
The sock API uses us, but we can just multiply